### DISCUSSION TITLE
Exception handling when enqueueing jobs

### DISCUSSION THREAD

**Discussion by matijaparavac** [jobrunr/jobrunr#622] [Q&A]
How to properly handle exception handling in child jobs?
Should we re-throw exception back to the parent job?(we use catch block for logging)
If yes will main/parent job have a retry policy then? Should we set retry on parent job to 0?

For example we have 100 enqueued jobs which handle transactions and if 1 of them fails even with retries what would happen?
Will the parent job fail? If ti failes, will it retry only this failed enqueued-job or all of them again?

**Comment by rdehuyss** [comment]
First question: Hi @matijaparavac - first a question: are you using JobRunr Pro?

**Reply by matijaparavac** [reply]
@rdehuyss No, I'm using free version. Thanks for answering!

**Reply by rdehuyss** [reply]
Then I don't understand what you mean as parent / child jobs are only available in the Pro version.

**Reply by matijaparavac** [reply]
@rdehuyss  Sorry didn't know this. By parent job I mean the long running job that is enqueuing other job:         
`BackgroundJob.<ComplianceJobService, Integer>enqueue(transactions.parallelStream(), (complianceJobService, transactionId) -> complianceJobService.saveComplianceForTransaction(transactionId, request));`

By child job I mean that saveComplianceForTransaction Job that is enqued.
I hope it is more clear now.

**Reply by rdehuyss** [reply]
These are 2 different jobs that don't know anything from each other - there isn't even a parent job, just a job that schedules other jobs.

**Reply by matijaparavac** [reply]
@rdehuyss Ok...I will attempt to rephrase question one more time:
Should we re-throw exception in job that schedules 100 other jobs?
Does that job have a retry policy then? Should we set retry on that job to 0?
If 1 of those 100 other enqueued jobs fails what would happen?
Will job that schedules other jobs use its retry policy and run through all those 100 scheduled jobs again?

**Reply by rdehuyss** [reply]
@matijaparavac please see https://www.jobrunr.io/en/. If that's not enough, feel free to reach out to take a support contract.
